amr-wind/wind_energy/ABLWallFunction.cpp

```diff
@@ -170,11 +170,18 @@ void ABLVelWallFunc::wall_model(
 
     amrex::Orientation zlo(amrex::Direction::z, amrex::Orientation::low);
     amrex::Orientation zhi(amrex::Direction::z, amrex::Orientation::high);
-    if ((velocity.bc_type()[zlo] != BC::wall_model) &&
-        (velocity.bc_type()[zhi] != BC::wall_model)) {
+    if (velocity.bc_type()[zhi] == BC::wall_model) {
+        amrex::Abort("ABL wall models are not applicable to a zhi BC");
+    }
+    if (velocity.bc_type()[zlo] != BC::wall_model) {
         return;
     }
 
+    if ((velocity.bc_type()[zhi] == BC::wall_model) &&
```

Comment:
this branch is not needed anymore since there's a more general `Abort` above

Reply:
good catch